models.py

```diff
@@ -300,122 +356,226 @@ def vdcnn(embedding_size,
         else:
-            x = _vdcnn_block(filter_nr, kernel_size, use_batch_norm, use_prelu, dropout_convo, l2_reg_convo,
-                             last_block=True)(x)
-
-    x = GlobalMaxPool1D()(x)
-    for i in range(repeat_dense):
-        x = _dense_block(dense_size, use_batch_norm, use_prelu, dropout_dense, l2_reg_dense)(x)
-    predictions = Dense(6, activation='sigmoid')(x)
+            x = _vdcnn_block(filter_nr, kernel_size, use_batch_norm, use_prelu, conv_dropout, dropout_mode,
+                             conv_kernel_reg_l2, conv_bias_reg_l2, batch_norm_first, last_block=True)(x)
+
+    predictions = _classification_block(dense_size=dense_size, repeat_dense=repeat_dense,
+                                        max_pooling=max_pooling,
+                                        mean_pooling=mean_pooling,
+                                        weighted_average_attention=weighted_average_attention,
+                                        concat_mode=concat_mode,
+                                        dropout=dense_dropout,
+                                        kernel_reg_l2=dense_kernel_reg_l2, bias_reg_l2=dense_bias_reg_l2,
+                                        use_prelu=use_prelu, use_batch_norm=use_batch_norm,
+                                        batch_norm_first=batch_norm_first)(x)
     model = Model(inputs=input_text, outputs=predictions)
     return model
 
 
-def _bn_relu_dropout_block(use_batch_norm, use_prelu, dropout):
+def _classification_block(dense_size, repeat_dense,
+                          max_pooling, mean_pooling, weighted_average_attention, concat_mode,
+                          dropout,
+                          kernel_reg_l2, bias_reg_l2,
+                          use_prelu, use_batch_norm, batch_norm_first):
+    def f(x):
+        if max_pooling:
+            x_max = GlobalMaxPool1D()(x)
+        else:
+            x_max = None
+
+        if mean_pooling:
+            x_mean = GlobalAveragePooling1D()(x)
+        else:
+            x_mean = None
+        if weighted_average_attention:
+            x_att = AttentionWeightedAverage()(x)
+        else:
+            x_att = None
+
+        x = [xi for xi in [x_max, x_mean, x_att] if xi is not None]
+        if len(x) == 1:
+            x = x[0]
+        else:
+            if concat_mode == 'concat':
+                x = concatenate(x, axis=-1)
+            else:
+                NotImplementedError('only mode concat for now')
+
+        for _ in range(repeat_dense):
+            x = _dense_block(dense_size=dense_size,
+                             use_batch_norm=use_batch_norm,
+                             use_prelu=use_prelu,
+                             dropout=dropout,
+                             kernel_reg_l2=kernel_reg_l2,
+                             bias_reg_l2=bias_reg_l2,
+                             batch_norm_first=batch_norm_first)(x)
+
+        x = Dense(6, activation="sigmoid")(x)
```

Comment:
This should be parametrized, or moved outside, so that User can specify both arguments.
Just like `def _convolutional_block` which is fully parametrized.